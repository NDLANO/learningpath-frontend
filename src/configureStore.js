/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { compose, createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import persistState from 'redux-localstorage';
import { routerMiddleware } from 'react-router-redux';

import reducers from './reducers';
import { errorReporter } from './middleware';

const slicer = paths =>
  // custom slicer because default slicer does not store falsy values
  state =>
    paths.reduce((acc, path) => {
      // eslint-disable-next-line no-param-reassign
      acc[path] = state[path];
      return acc;
    }, {});

export default function configureStore(initialState, history) {
  const middleware = routerMiddleware(history);
  const createFinalStore = compose(
    applyMiddleware(thunkMiddleware, errorReporter, middleware),
    __CLIENT__
      ? persistState(['authenticated', 'accessToken', 'user', 'accessToken'], {
          key: 'ndla:sti',
          slicer,
        })
      : f => f,
    __CLIENT__ && window && window.devToolsExtension
      ? window.devToolsExtension()
      : f => f,
  )(createStore);

  return createFinalStore(reducers, initialState);
}
