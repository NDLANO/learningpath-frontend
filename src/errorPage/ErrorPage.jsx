/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from "react";
import PropTypes from "prop-types";
import { Wrapper, OneColumn, Footer } from "../common/Layout";
import Logo from "../common/Logo";
import { DefaultErrorMessage } from "./DefaultErrorMessage";

const ErrorPage = ({ locale }) => (
  <Wrapper>
    <OneColumn>
      <div className="masthead_wrapper">
        <div className="masthead">
          <div className="masthead_left--desktop">
            <Logo />
          </div>
        </div>
      </div>
      <DefaultErrorMessage />
    </OneColumn>
    <Footer locale={locale} />
  </Wrapper>
);

ErrorPage.propTypes = {
  locale: PropTypes.string.isRequired,
};

export default ErrorPage;
