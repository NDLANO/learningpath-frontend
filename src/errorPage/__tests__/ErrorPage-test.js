/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @jest-environment jsdom
 */

import React from 'react';
import { StaticRouter } from 'react-router';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import ErrorPage from '../ErrorPage';
import configureStore from '../../configureStore';

jest.mock('../../config', () => ({
  zendeskWidgetKey: '123',
}));

test('ErrorPage renderers correctly', () => {
  const store = configureStore({ locale: 'nb' });
  const component = renderer.create(
    <Provider store={store} locale="nb">
      <StaticRouter>
        <ErrorPage locale="nb" />
      </StaticRouter>
    </Provider>,
  );

  expect(component.toJSON()).toMatchSnapshot();
});
