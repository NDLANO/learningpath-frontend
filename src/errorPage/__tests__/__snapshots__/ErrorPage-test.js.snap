// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage renderers correctly 1`] = `
.emotion-0 {
  display: inline-block;
  color: #ffffff;
  background-color: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  padding: 4px 13px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.625;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 700;
  -webkit-transition: all .2s cubic-bezier(0.17,0.04,0.03,0.94);
  transition: all .2s cubic-bezier(0.17,0.04,0.03,0.94);
  color: #20588f;
  background-color: transparent;
  border: 2px solid #20588f;
  font-weight: 700;
  box-shadow: none;
  border-radius: 2px;
  background-color: white;
  float: right;
}

.emotion-0:hover,
.emotion-0:focus {
  color: white;
  background-color: #184673;
  border: 2px solid rgba(32,88,143,0);
  -webkit-transform: translateY(1px) translateX(1px);
  -ms-transform: translateY(1px) translateX(1px);
  transform: translateY(1px) translateX(1px);
}

.emotion-0:active {
  -webkit-transform: translateY(2px) translateX(2px);
  -ms-transform: translateY(2px) translateX(2px);
  transform: translateY(2px) translateX(2px);
}

.emotion-0[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
  -webkit-transform: translateY(0) translateX(0);
  -ms-transform: translateY(0) translateX(0);
  transform: translateY(0) translateX(0);
}

.emotion-0:focus {
  box-shadow: 0 0 2px #20588f;
}

.emotion-0:hover,
.emotion-0:focus {
  color: white;
  background-color: #20588f;
  border: 2px solid transparent;
  -webkit-transform: translateY(0) translateX(0);
  -ms-transform: translateY(0) translateX(0);
  transform: translateY(0) translateX(0);
}

.emotion-0:disabled {
  border: 2px solid transparent;
  cursor: not-allowed;
  -webkit-transform: translateY(0) translateX(0);
  -ms-transform: translateY(0) translateX(0);
  transform: translateY(0) translateX(0);
}

@media (min-width:61.3125em) {
  .emotion-0 {
    position: fixed;
    right: 2rem;
    bottom: 2rem;
    padding: 0.4rem 0.6rem;
  }
}

<div
  className="wrapper"
>
  <div
    className="one-column"
  >
    <div
      className="masthead_wrapper"
    >
      <div
        className="masthead"
      >
        <div
          className="masthead_left--desktop"
        >
          <div
            className="logo un-button"
          >
            <a
              className="logo_link"
              href="/"
              onClick={[Function]}
            >
              Nasjonal digital læringsarena
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="infoblock-error"
    >
      <img
        alt="Noe gikk galt."
        role="presentation"
        src="/oops.gif"
        style={
          Object {
            "maxWidth": "100%",
          }
        }
      />
      <div>
        <h1>
          Ops, noe gikk galt
        </h1>
        <span>
          Beklager, en feil oppsto.
        </span>
      </div>
    </div>
  </div>
  <footer
    className="footer"
  >
    <form
      className="footer_language-form"
    >
      <label
        className="footer_language-label footer--bold"
        htmlFor="language-select"
      >
        Velg språk (language): 
      </label>
      <select
        data-cy="select-locale"
        onChange={[Function]}
        value="nb"
      >
        <option
          value="nb"
        >
          Bokmål
        </option>
        <option
          value="nn"
        >
          Nynorsk
        </option>
        <option
          value="en"
        >
          English
        </option>
      </select>
    </form>
    <ul
      className="footer_list footer--bold"
    >
      <li
        className="footer_item"
      >
        <a
          className="footer_link"
          data-cy="footer-about-ndla"
          href="https://om.ndla.no/"
        >
          Om NDLA
        </a>
      </li>
    </ul>
    <div
      className="footer_ruler"
    />
    <p
      className="footer_text"
    >
      <span
        className="footer_editor"
      >
        Ansvarlig redaktør: 
        <strong>
          Sigurd Trageton
        </strong>
      </span>
    </p>
    <button
      className="emotion-0 emotion-1"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "bottom": "5.5rem",
        }
      }
      type="button"
    >
      Spør NDLA
    </button>
    <p
      className="footer_text"
    >
      Nettstedet er utarbeidet av NDLA som åpen kildekode.
    </p>
  </footer>
</div>
`;
